Implement restaurant list

canary4651 opened the pull request:
일단 pr은 빨리 보내고..! 
강의 보고 또 참고해야하는 문서들 보고 공부하고
내일 계속 commit 올리겠습니다!

canary4651 committed:
Implement restaurant list

Add base code

Kihyun92 commented:
좋습니다! 다음 커밋을 기다릴게요~ 😊

canary4651 committed:
Put button to APP and test

canary4651 committed:
Add region buttons

canary4651 committed:
Add category buttons and test

Kihyun92 reviewed (changes requested):
차근차근 진행해보고 계신 것 같아요 👍🏼 

다만 커밋의 단위를 조금 쪼개보면 어떨까요?

TDD cycle을 지키며 작업중이신 것 같지만, 몇주만 훈련이라 생각하고 커밋 단위를 `red-green-refactoring`이 되도록 해보시면 좋을 것 같아요~ 저희 과제 제한 조건이기도 하구요 ㅎㅎ

![image](https://user-images.githubusercontent.com/35797540/108999242-617b7000-76e5-11eb-89b2-abad6638c325.png)

canary4651 commented:
> 차근차근 진행해보고 계신 것 같아요 👍🏼
> 
> 다만 커밋의 단위를 조금 쪼개보면 어떨까요?
> 
> TDD cycle을 지키며 작업중이신 것 같지만, 몇주만 훈련이라 생각하고 커밋 단위를 `red-green-refactoring`이 되도록 해보시면 좋을 것 같아요~ 저희 과제 제한 조건이기도 하구요 ㅎㅎ
> 
> ![image](https://user-images.githubusercontent.com/35797540/108999242-617b7000-76e5-11eb-89b2-abad6638c325.png)

아하 네!! 더 신경써서 쪼개서 커밋 보내겠습니다.

canary4651 committed:
🔴 Create Regions component test

canary4651 committed:
🟢 Create Regions component

canary4651 committed:
♻️  Refactoring Regions component

canary4651 committed:
🔴 Create RegionsContianer test

canary4651 committed:
🟢 Create RegionsContainer component

canary4651 committed:
♻️  Refactoring RegionsContanier and add reducer

canary4651 committed:
🔴 Create reducer test

canary4651 committed:
🟢 Create reducer component

canary4651 committed:
♻️  Refactoring App and RegionsCOntainer test

canary4651 committed:
🔴 Create setRegions reducer test

canary4651 committed:
🟢 Create setRegions action and reducer

canary4651 committed:
♻️  Refactoring App test and App

Kihyun92 commented on src/Regions.jsx:
```diff
@@ -0,0 +1,17 @@
+  return (
+    <div>
+      <ul>
+        {regions.map((region) => (
```
구조 분해 할당으로 처리해줘도 좋을 것 같네요 :)

Kihyun92 commented on src/RegionsContainer.jsx:
```diff
@@ -0,0 +1,17 @@
+  }));
+
+  return (
+    <div>
```
div 태그로 감싸줄 필요가 있을까요? 특별한 이유가 없다면 제거해도 좋을 것 같아요 ㅎㅎ

Kihyun92 commented on src/reducer.js:
```diff
@@ -0,0 +1,15 @@
+};
+
+export default function reducer(state = initialState, action = { type: '' }) {
+  if (action.type === 'setRegions') {
+    const { regions } = action.payload;
+    return {
+      ...state,
+      regions,
+    };
+  }
```
지금은 괜찮지만 이번 과제에서도 `if`나 `switch`를 사용하지 않고 해결해주세요 :)

Kihyun92 reviewed (changes requested):
커밋 메시지를 보니 `TDD cycle`을 잘 지키려 노력해보신 것 같아요 👍🏼 

잘 진행중이신 것 같은데, 혹시 의논이 필요한 부분이 있다면 편하게 말씀해주세요~ 😊

canary4651 committed:
♻️  Refactoring reduer

Kihyun92 reviewed (commented):
아직 진행중이셔서 리뷰 드릴 부분이 많지 않네요 ㅠㅠ

차근차근 진행하시면서 API 호출해서 데이터를 가져오는 부분도 처리해보시면 좋을 것 같아요~

화이팅입니다!

canary4651 committed:
🔴 Create loadDefaultDate and fetchRegions

canary4651 committed:
🟢 Make loadDefaultDate and fetchRegions

Install `redux-thunk`.

canary4651 commented:
> 아직 진행중이셔서 리뷰 드릴 부분이 많지 않네요 ㅠㅠ
> 
> 차근차근 진행하시면서 API 호출해서 데이터를 가져오는 부분도 처리해보시면 좋을 것 같아요~
> 
> 화이팅입니다!

앗 네네 ㅠㅠ 제가 이번주는 진행속도가 많이 느리네요... 지금부터 계속 해보고 있습니다..!

canary4651 committed:
🔴 Create selectRegions

canary4651 committed:
🟢 Create selectRegion

canary4651 committed:
♻️  Refactoring selectedRegion

Make "V" mark on the selected region button.

Kihyun92 commented on src/App.jsx:
```diff
@@ -1,7 +1,35 @@
+      <h1>Restaurant List</h1>
+      <RegionsContianer />
+      <ul>
+        {categories.map((category) => (
```
구조분해할당으로 처리해줘도 좋겠네요~

canary4651 committed:
🔴 Create CategoriesContainer

Add categories component: `Categoreis` and `CategoriesContainer`

Add reducers and actions about categories.

Kihyun92 commented on src/Regions.jsx:
```diff
@@ -0,0 +1,21 @@
+          <li key={id}>
+            <button
+              type="button"
+              onClick={() => onClick(name)}
```
클릭시 해당 네이밍을 넘기는거도 좋지만, 변할 가능성이 있는 값 같아요~

DB에서 변하지 않음을 보장하는 id 값으로 처리해주는건 어떨까요?

canary4651 replied:
네 반영했습니다!

canary4651 replied:
앗 제가.... id로 넘겼다가 api에서 계속 숫자로 저장되는 이슈가 발생해서 넘어가지지 않더라구요 ㅠㅠ 
그래서 다시 name으로 받게 변경했습니다.

Kihyun92 commented on src/actions.js:
```diff
@@ -0,0 +1,25 @@
+  };
+}
+
+export function loadDefaultData() {
```
`defaultData` 라는 네이밍은 명확하지 않은 것 같아요.

어떤 값을 로딩해오는건지 알 수 있게 표현해주면 어떨까요?

canary4651 replied:
regions 들과 categories들의 목록을 들고 있는 데... regions랑 categories를 합치면 너무 길어질거같아서 고민입니다. 
`loadListData` 요런식으로 해도 어떤 값인지 모를 거 같은데, 좋은 네이밍이 있을까요?

Kihyun92 replied:
아하 최초 초기값을 가져올때만 사용하는 경우라면 loadInitialData 로 해줘도 좋을 것 같네요 :)

Kihyun92 reviewed (changes requested):
꾸준히 진행해보시는게 좋네요 👍🏼  화이팅입니다 ㅎㅎ

올라온 강의 풀이 영상을 본 뒤에 진행해보셔도 좋을 것 같아요 :)

canary4651 committed:
🟢 Create fetchCategories and fix typo

canary4651 committed:
🔴 Create RestaurnatContainer

canary4651 committed:
🟢 Make fetchRestaurnats

canary4651 committed:
♻️  Refactoring fetchRestaurants

canary4651 committed:
♻️  Refactoring loadRestaurants and api

canary4651 committed:
♻️  Create action and api test

canary4651 committed:
♻️  Refactoring Restaurants Component

Seperate `Restaurnats` and `RestaurantsContainer`.

canary4651 committed:
WIP...

canary4651 committed:
♻️  Refactoring API Issue

Fix region handler according to the api spec.

canary4651 committed:
♻️  Pass all test!

Add loadRestaurants action to `regionsContainer`.

Kihyun92 reviewed (approved):
고생하셨습니다~ 쉽지않은 과제였는데 잘 진행해주신 것 같아요 :)

이번 과제를 하면서 무엇을 배웠나요? 더 배우고 싶은 부분이 있나요? ㅎㅎ

강의 풀이 영상 꼭 보시면 좋을 것 같습니다~

슬랙 채널에 회고 남겨주시면 머지하도록 하겠습니다 :)
